src/shared/stores/sync-errors-store.js

```diff
@@ -0,0 +1,66 @@
+import create from 'zustand'
+import {
+    parseCellId,
+    getCellIdFromMutationKey,
+    getCellIdFromDataValueParams,
+} from './get-cell-id.js'
+
+const inititalState = {
+    // keyed by cellId, see getCellId
+    errors: {},
+}
+
+export const useSyncErrorsStore = create((set, get) => ({
+    ...inititalState,
+    setErrorById: (cellId, error) =>
+        set((state) => {
+            const errors = { ...state.errors }
+            errors[cellId] = error
+            return { errors }
+        }),
+    setErrorByMutationKey: (mutationKey, error) => {
+        const cellId = getCellIdFromMutationKey(mutationKey)
+        get().setErrorById(cellId, error)
+    },
+    setError: (apiMutationError) => {
+        const cellId = getCellIdFromMutationKey(apiMutationError.mutationKey)
+        get().setErrorById(cellId, apiMutationError)
+    },
+
+    clearErrorById: (cellId) =>
+        set((state) => {
+            const errors = { ...state.errors }
+            delete errors[cellId]
+            return { errors }
+        }),
+    clearErrorByMutationKey: (mutationKey) => {
+        const cellId = getCellIdFromMutationKey(mutationKey)
+        get().clearErrorById(cellId)
+    },
+    clearErrorByDataValueParams: (dataValueParams) => {
+        const cellId = getCellIdFromDataValueParams(dataValueParams)
+        get().clearErrorById(cellId)
+    },
+    clearAll: () => set(inititalState),
+
+    getErrors: () => get().errors,
+    getErrorById: (cellId) => get().getErrors()[cellId],
+    getErrorByDataValueParams: (dataValueParams) =>
+        get().getErrorById(getCellIdFromDataValueParams(dataValueParams)),
+    getErrorByMutationKey: (mutationKey) =>
```

Comment:
No, but I think it could be valuable, and "completes" the API. I think I did use this in a iteration, but then created `getErrorByDataValueParams`-instead - since that was easily available in the component. 
Generally I tend to keep things like that, since it can easily be usable in the future, and it makes a more "complete" API for the store. However, I do understand this is not a library - so I can remove if you think having remnants like this are unnecessary. But you can see below that `getErrorsForSelection` and unused, and then we end up using them.

Reply:
Up to you, but I think it's generally easier to get rid of code we're not using (or at least comment it out), so that it doesn't end up causing confusion in the future (/being maintained needlessly)